### ISSUE TITLE
[REQUEST] Add -g as default flag for tcc

### ISSUE THREAD

**Issue by pervognsen** [compiler-explorer/compiler-explorer#2748]
**Is your feature request related to a problem? Please describe.**

If you use tcc as a compiler, you currently don't get to see symbol names in the disassembly or the colored line correlation highlighting. This makes the disassembly hard to follow for humans and greatly reduces the value provided by Compiler Explorer's special UI features.

**Describe the solution you'd like**

Add -g as a default command line flag for tcc to enable debug info.

**Additional context**

You can see a before/after example here: https://godbolt.org/z/r9T7zPh4b

**Comment by RubenRBS** [comment]
Hi! Thanks for taking the time to open up a request.

By the looks of it, this seems like intended behaviour, based on #2365, which implemented it and left `-g` out. @partouf might have some more insight as to how different tcc's is to the usual one and what might be done here

**Comment by pervognsen** [comment]
That issue says that -g "means something different" with tcc. It's true that tcc's -g debug info is more shallow than the likes of gcc and clang but it's the only way to get named symbols and line numbers in the ELF/COFF file with tcc. You can see it working in the before/after example I linked.

**Comment by partouf** [comment]
I don't really remember -g working this well in the past, but I only see 1 commit in trunk that might have something to do with the dwarf outputs.

Either way, it seems to make sense to use -g as a default now.

**Comment by partouf** [comment]
TLDR: Let's just do it
